monkey/monkey_island/cc/repositories/agent_plugin_repository_caching_decorator.py

```diff
@@ -0,0 +1,18 @@
+from functools import lru_cache
+
+from common.agent_plugins import AgentPlugin, AgentPluginType
+
+from . import IAgentPluginRepository
+
+
+class AgentPluginRepositoryCachingDecorator(IAgentPluginRepository):
+    """
+    An IAgentPluginRepository decorator that provides caching for other IAgentPluginRepositories
+    """
+
+    def __init__(self, agent_plugin_repository: IAgentPluginRepository):
+        self._agent_plugin_repository = agent_plugin_repository
+
+    @lru_cache(maxsize=16)
```

Comment:
I'd suggest to extract this number as a `PLUGIN_CACHE_SIZE` or something similar, to make it easier to find.

Reply:
We don't even need to set a maxsize for now.